src/mlpack/methods/ann/layer/neural_turing_machine.hpp

```diff
@@ -0,0 +1,256 @@
+/**
+ * @file neural_turing_machine.hpp
+ * @author Sumedh Ghaisas
+ *
+ * Definition of the Neural Turing Machine class.
+ *
+ * //TODO add reference
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_NTM_HPP
+#define MLPACK_METHODS_ANN_LAYER_NTM_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include <boost/ptr_container/ptr_vector.hpp>
+
+#include "../visitor/delta_visitor.hpp"
+#include "../visitor/output_parameter_visitor.hpp"
+#include "../visitor/forward_with_memory_visitor.hpp"
+#include "../activation_functions/softplus_function.hpp"
+
+#include "layer_types.hpp"
+#include "add_merge.hpp"
+#include "sequential.hpp"
+
+#include "memory_head.hpp"
+#include "read_memory.hpp"
+#include "write_memory.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * An implementation of Neural Turing Machine Layer.
```

Comment:
Yup. I planned to do that with the TODO of reference. But got very excited with the working version and started testing it right away.